### PR TITLE
Rewrite the input interface of model initialization and add training process visualization code to facilitate the training process

### PR DESCRIPTION
**Pull request by Mingbo-Lee**
I replaced numeric assignments with variables assignments for the model initialization part of split learning, so that as long as the data is replaced, the tutorial can be quickly applied to other data, and I add visualization code to visualize the training process, so that the model training situation can be quickly perceived.

### CODE REVIEW

**Comment by zhouaihui** [comment]
hi, @Mingbo-Lee , can you create an issue for this PR?

**Comment by Mingbo-Lee** [comment]
Thank you for your suggestion. @zhouaihui

**Comment by Mingbo-Lee** [comment]
The issue was created: https://github.com/secretflow/secretflow/issues/431

File: docs/tutorial/Split_Learning_for_bank_marketing.ipynb [hunk 1]
```diff
@@ -1520,6 +1106,133 @@
...
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
```
Comment (zhangxingmeng) [inline docs/tutorial/Split_Learning_for_bank_marketing.ipynb:1125]: The metrics here must be consistent with the results in the `summary`. You need to modify the results below in the summary as well

Comment (zhangxingmeng) [reply docs/tutorial/Split_Learning_for_bank_marketing.ipynb:1125]: Epochs in standalone version be consistent with those in the federated version

**Comment by Mingbo-Lee** [comment]
I have modified the jupyter notebook **summary** as you suggested. Please review. Thank you! @zhangxingmeng

File: docs/tutorial/Split_Learning_for_bank_marketing.ipynb [hunk 2]
```diff
@@ -1615,36 +1773,49 @@
...
       "  alice_data.loc[:, 'education'] = encoder.fit_transform(alice_data['education'])\n",
-      "/tmp/ipykernel_3681029/3166940482.py:10: SettingWithCopyWarning: \n",
+      "/tmp/ipykernel_818514/593075830.py:9: DeprecationWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`\n",
+      "  alice_data.loc[:, 'education'] = encoder.fit_transform(alice_data['education'])\n",
+      "/tmp/ipykernel_818514/593075830.py:10: SettingWithCopyWarning: \n",
       "A value is trying to be set on a copy of a slice from a DataFrame.\n",
       "Try using .loc[row_indexer,col_indexer] = value instead\n",
       "\n",
       "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
+      "  alice_data.loc[:, 'y'] =  encoder.fit_transform(alice_data['y'])\n",
```
[the remark this answers is not in the document]
Comment (Mingbo-Lee) [reply docs/tutorial/Split_Learning_for_bank_marketing.ipynb:1817]: I modified the code as you suggested,:
```
encoder = LabelEncoder()
alice_data['job'] = encoder.fit_transform(alice_data['job'])
alice_data['marital'] = encoder.fit_transform(alice_data['marital'])
alice_data['education'] = encoder.fit_transform(alice_data['education'])
alice_data['y'] =  encoder.fit_transform(alice_data['y'])
```
However,  the message of warning  of  SettingWithCopyWarning  still appeared:
```
/tmp/ipykernel_13988/1420406069.py:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  alice_data['job'] = encoder.fit_transform(alice_data['job'])
/tmp/ipykernel_13988/1420406069.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  alice_data['marital'] = encoder.fit_transform(alice_data['marital'])
/tmp/ipykernel_13988/1420406069.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  alice_data['education'] = encoder.fit_transform(alice_data['education'])
/tmp/ipykernel_13988/1420406069.py:10: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  alice_data['y'] =  encoder.fit_transform(alice_data['y'])
```
Do you have another solution? or could you consider to approve this pull request and we plan to solve the warning of code later in another pull request?

[the remark this answers is not in the document]
Comment (zhangxingmeng) [reply docs/tutorial/Split_Learning_for_bank_marketing.ipynb:1817]: This problem is caused by upgrade of pandas, and I have no better solution for the moment. 
Operating numpy seems a workaround solution. You can try this
```
encoder = LabelEncoder()
alice_data.loc[:, 'job'].values[:] = encoder.fit_transform(alice_data['job'])
```

[the remark this answers is not in the document]
Comment (Mingbo-Lee) [reply docs/tutorial/Split_Learning_for_bank_marketing.ipynb:1817]: Oh, as you suggested to run the code,  more serious bug happened which made the code does not run.

```
ValueError                                Traceback (most recent call last)
Cell In[38], line 1
----> 1 history =single_model.fit(train_data,train_label,validation_data=(test_data,test_label),batch_size=128,epochs=10,shuffle=False)

File ~/.conda/envs/sf/lib/python3.8/site-packages/keras/utils/traceback_utils.py:70, in filter_traceback.<locals>.error_handler(*args, **kwargs)
     67     filtered_tb = _process_traceback_frames(e.__traceback__)
     68     # To get the full stack trace, call:
     69     # `tf.debugging.disable_traceback_filtering()`
---> 70     raise e.with_traceback(filtered_tb) from None
     71 finally:
     72     del filtered_tb

File ~/.conda/envs/sf/lib/python3.8/site-packages/tensorflow/python/framework/constant_op.py:102, in convert_to_eager_tensor(value, ctx, dtype)
    100     dtype = dtypes.as_dtype(dtype).as_datatype_enum
    101 ctx.ensure_initialized()
--> 102 return ops.EagerTensor(value, ctx.device_name, dtype)

ValueError: Failed to convert a NumPy array to a Tensor (Unsupported object type int).
```
I will try to find another solution.

**Comment by Mingbo-Lee** [comment]
Hi! @zhangxingmeng  The warnning message of pandas has be solved. could you please review again?

**Review by zhangxingmeng — APPROVED** [review]
LGTM

**Pull request merged**